adding the social grade and the personal students preferred students.

noy248 opened the pull request:
Student page with full parameters view and editing.
adding the social grading and personal preffered of the students

noy248 committed:
adding the social grade and the personal students preferred students.

edenbs commented on client/app/students/new-student/new-student.html:
```diff
@@ -5,6 +5,7 @@
         </div>
     </md-toolbar>
     <form name="addStudentForm">
+       <div>
```
Why did you add this empty div? it doesn't seem to do anything
Delete it if it's unnecessary

noy248 replied:
It was to separate the form from the save button, without it when adding new values to fill in creating new student the save button will stay in the middle of the form.

edenbs commented on client/app/students/new-student/new-student.html:
```diff
@@ -33,9 +34,36 @@
+           </div>
         </md-dialog-contnet>
+       </div>
+        <div>
```
Same as above

edenbs commented on client/app/students/new-student/new-student.html:
```diff
@@ -33,9 +34,36 @@
             </md-input-container>
+            <md-input-container>
+                <label>Social Grade</label>
+                <input name="social" ng-model="student.social"  ng-pattern='/^([1-4])$/' required/>
```
Maybe this should be `type="number"`? it will automatically convert it to a number

noy248 replied:
changed at new-student.html and in the edit option at student.html.

edenbs commented on server/api/student/student.model.js:
```diff
@@ -43,6 +45,18 @@ const studentSchema = new Schema({
         max: 100
+    },
+    social: {
+        enum:socialgrades,
```
Does it work with enum AND type number? isn't type number sufficient?

noy248 replied:
Right  the type should be enough with the min,max attributes because its numeric.
I deleted the enum reference for the next commit.

edenbs reviewed (commented):
Overall looks really good :)
The only big issue is that I'm thinking maybe we should use autocomplete instead of a regular input for the student preferations, it will be easier for teachers to look for students instead of entering their id. Meaning that in the input they will enter the name (the autocomplete will show them all the students filtered by the name and their id in brackets), and when they select the student, it will still show the student name but the ng-model value will be the _id.
Tell me what you think about it

https://material.angularjs.org/latest/demo/autocomplete

noy248 committed:
adding the loading of the excel file.

there are comment in the client and server side of unsolved problems.
1) The promise in the upload-students doesn't work from ng-upload-file Upload function. - so there is no proper error handling
2) there is no TEMPLATE file for the user.
3) There is not yet a solution for student who is already in the DB in other school - needs to change students constraint to id + school.

edenbs commented on client/app/students/students.controller.js:
```diff
@@ -132,6 +132,10 @@ angular.module('classify').controller('StudentsController', function($scope, $md
     };
 
+    $scope.isEditor = function () {
+            return auth.hasRole('editor');
```
why does the indents look weird?

edenbs commented on client/app/students/students.html:
```diff
@@ -84,7 +84,12 @@
                     <td md-cell ng-if="!isEditor()">{{ item.social }}</td>
-                    <td md-cell ng-if="isEditor()" ng-click="edit($event, item, 'social')" >{{ item.social }} </td>
+                    <td md-cell ng-if="isEditor()">
+                        <md-input-container>
```
You should use edit inside the cell rather than put a simple input.
See that you can add your constraints like min max in the controller inside the $scope.validators (I see that you did, you should use it with the edit function)

edenbs commented on client/app/students/students.controller.js:
```diff
@@ -136,7 +146,29 @@ angular.module('classify').controller('StudentsController', function($scope, $md
 
-    $scope.isEditor = function () {
-        return auth.hasRole('editor');
+    $scope.changeSocial = function (student) {
```
see comment below -
I think this is unnecessary. edit should be enough :)

edenbs commented on client/app/students/students.html:
```diff
@@ -80,6 +83,17 @@
                     <td md-cell ng-if="!isEditor()">{{ item.avgGrade }}</td>
+                    <td md-cell ng-if="!isEditor()">{{ item.social }}</td>
+                    <td md-cell ng-if="isEditor()">
+                        <md-input-container>
```
You should use edit inside the cell rather than put a simple input.
See that you can add your constraints like min max in the controller inside the $scope.validators (I see that you did, you should use it with the edit function)

noy248 replied:
The constraint of the value allow user to get every value on the input, but min/max  
doesn't give the option to other than 1-4.

edenbs commented on client/index.html:
```diff
@@ -12,11 +12,14 @@
     <link rel="stylesheet" href="bower_components/angular-hamburglar/dist/hamburglar.min.css" />
     <link rel="stylesheet" href="bower_components/mdi/css/materialdesignicons.css" />
     <link rel="stylesheet" href="bower_components/mdi/css/materialdesignicons.min.css" />
+    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
```
This shouldn't be inside the bower tag, move it two lines below

edenbs commented on package.json:
```diff
@@ -56,7 +57,7 @@
     "passport-local-mongoose": "^4.4.0",
     "pify": "^3.0.0",
-    "xlsx": "^0.12.7",
+    "xlsjs": "^1.0.1",
```
On npm it says the package is deprecated and we should use xlsx, use it if possible
https://www.npmjs.com/package/xlsjs

edenbs commented on server/api/student/index.js:
```diff
@@ -2,11 +2,13 @@ import {AsyncRouter} from 'express-async-router';
 import paginate from 'express-paginate';
-
+import multiparty from 'connect-multiparty';
+var multipartyMiddleware = multiparty();
```
Use const for variables that are not gonna change
In ES6 in general we don't use `var` but  `const` and `let`

edenbs commented on server/api/student/student.model.js:
```diff
@@ -43,6 +43,17 @@ const studentSchema = new Schema({
+        max: 4
+    },
+    prefer: {
+        type:{
```
If the label was meant to represent hard coded 'first', 'second'...
I think it's better to actually create the prefer object like this:
```js
prefer: {
  first: {
    id: String
  },
 second: {
  },
 third: {
  }
}
```

noy248 committed:
adding the students preferences
